cli/server.go

```diff
@@ -395,6 +395,22 @@ func (r *RootCmd) Server(newAPI func(context.Context, *coderd.Options) (*coderd.
 
 			config := r.createConfig()
 
+			if vals.PostgresURL == "" && vals.PostgresHost != "" && vals.PostgresUsername != "" && vals.PostgresPassword != "" && vals.PostgresDatabase != "" {
+				pgURL := fmt.Sprintf("postgres://%s:%s@%s:%s/%s?%s",
```

Comment:
Consider what happens if only username is given, or no database, etc. I'm not sure if `Set` below will error under what circumstances, but is the error message helpful in the user figuring out what's missing? Additional validation needed?

Reply:
@mafredri 
if either of them is not set, the existing behaviour will continue, we will only form the URL if `PostgresHost` `PostgresUsername` `PostgresPassword` and `PostgresDatabase` are present